cpp/src/Slice/Util.h

```diff
@@ -73,10 +73,18 @@ std::string camelCase(const std::string&);
 std::string pascalCase(const std::string&);
 std::string snakeCase(const std::string&);
 
-// Returns a new list of data members sorted as follows:
-// - non-tagged data members listed first but kept in the same order
-// - tagged data members listed last and sorted in tag order
-DataMemberList sortForMarshaling(const DataMemberList&);
+// Sorts a list of members in place as follows:
+// - non-tagged members listed first but kept in the same order
+// - tagged members listed last and sorted in tag order
+void sortMembers(DataMemberList& members);
+// Returns a pair of lists corresponding to the outputs of sortedTaggedMembers and requiredMembers respectively.
+std::pair<DataMemberList, DataMemberList> getSortedMembers(const DataMemberList& members);
```

Comment:
Just pure convenience! the behavior of 'getSortedMembers' and 'sortMembers' is different, 'sortMembers' sorts a list into marshalling order in place, and most importantly it keeps everything in one list. 'getSortedMembers' actually separates the list into separate lists, one for non-tagged members, and another for tagged. This is useful since there's many places where we only want to iterate over required members, or tagged, but not both at the same time.

Originally I only have 'getSortedMembers', I added 'sortedTaggedMembers' and 'requiredMembers' since there were a few cases where we only cared about one or the other and didn't need both, so I added them in to avoid creating a list we didn't care about. Even though the comment says the outputs of 'getSortedMembers' correspond to the other two, this doesn't mean it just calls the two and wraps them together, it's a standalone method. The sorting order of the lists is identical is all and it was easier to write that it corresponds to them instead of re-writing the same thing I just already did.